File: frontend/src/app/cause.ts
import { Time } from '@angular/common';
import { User } from './user';

export class Cause {
    id: string;
    owner: string;
    name: string;
    location: string;
    description: string;
    photoPath: string;
    date: Date;
    time: number;
    commentIds: Array<string>;
    participantIds: Array<string>;
    causeType: string;

    constructor(jsonObj: any) {
        for (let prop in jsonObj) {
            this[prop] = jsonObj[prop];
        }
    }
}

File: backend/src/main/java/com/givers/repository/entity/Cause.java
package com.givers.repository.entity;

import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Cause {
	@Id
	private String id;
	private String name;
	private String owner;
	private String location;
	private String description;
	private String causeType;
	private String imagePath;
	private Long time;
	private List<String> commentIds;
	private List<String> participantIds;
}
